frontend/src/BinaryModifyView.svelte

```diff
@@ -181,7 +181,9 @@
         await selectedResource.queue_patch(patchData, startOffset, endOffset);
       }
 
-      resourceNodeDataMap[$selected].modified = true;
+      resourceNodeDataMap[$selected] = {
+        lastModified: true,
+      };
```

Comment:
Because of the reactive block linked below, if all of the values in the object are set to `undefined` (which they are here, except for `lastModified`), the values will be reverted to their defaults. This will, for example, collapse objects that may have been collapsed before. https://github.com/redballoonsecurity/ofrak/blob/eb0c5d6d2a3942abc11054f2b46f3a99757b37bd/frontend/src/ResourceTreeNode.svelte#L132-L158

My suggestion is to set the variable the way it was before with `.lastModified`, but wrap everything in a check to make sure the object is not `undefined` before you set its field.

```suggestion
      if (resourceNodeDataMap[$selected]) {
        resourceNodeDataMap[$selected].lastModified = true;
      } else {
        resourceNodeDataMap[$selected] = {
          lastModified: true,
        };
      }
```

This change will have to be made in a couple of places where this issue was allowed to slip through the cracks in the "run components in GUI" PR.

Reply:
As an alternative, there could be a helper method that does the setting. Something like this (haven't tested, just a draft off the dome).

``` javascript
function updateDataMap(id, key, value) {
  if (resourceNodeDataMap[id]) {
    resourceNodeDataMap[id][key] = value;
  } else {
    resourceNodeDataMap[id] = {key: value};
  }
}
```

Then that function could be called with `updateDataMap($selected, "lastModified", true)` in this case.

Reply:
I didn't wind up going with the helper function